docs/userguide/declarative_config.rst

```diff
@@ -192,28 +192,28 @@ obsoletes                                          list-comma
 Options
 -------
 
-=======================  ===================================  =============== =========
+=======================  ===================================  =============== ====================
 Key                      Type                                 Minimum Version Notes
-=======================  ===================================  =============== =========
+=======================  ===================================  =============== ====================
 zip_safe                 bool
-setup_requires           list-semi                            36.7.0
-install_requires         list-semi
-extras_require           section                                              [#opt-2]_
+setup_requires           file:, list-semi                     36.7.0          [#opt-5]_
```

Comment:
Question: would it make more sense to restrict this change to `install_requires` and `extras_requires`?

`tests_require` is deprecated, so I feel like we should not touch it.

`setup_requires` is largely obsoleted by PEP 517.
In very specific use cases people might still reach for it inside `setup.py`, but not in the context of `setup.cfg`.

I understand that adding support for the `file:` directive for `setup_requires` and `tests_require` basically come for free, but I still think we should refrain from doing that.

Reply:
I get where you're coming from here, but various comments in #1951 say `file:` support is the last thing preventing them from moving from an imperative `setup.py` to a PEP 518 `setup.cfg` build. So, if those folks had been using files for their `setup_requires` and `tests_require`, not allowing `file:` for them would make that porting work less straightforward.

I have no other strong opinion on this, just that thought :)

Reply:
Thank you very much @akx for pointing this out. I re-read the discussion in issue 1951 and I did not see any special request for `setup_requires` and `tests_requires`.

My view is that `tests_requires`, already does not work, so I don't think it would be helping anyone interested in the migration.

On the other hand, `setup_requires` only have value (right now) when computed dynamically via `setup.py`.

However I can see people starting to use it once it gets `file:` support just to workaround limitations in whichever GitHub bots they might be using for dependency management. In my opinion that would be an unfortunate side effect, since PEP 518 should be preferred.

---

On a side note, considering the forward compatibility with `pyproject.toml`:

- Currently `setup_requires` can be automatically converted to `[build-system] requires` by an automatic generator.
- If `setup_requires` gains the capability of being dynamically defined via `file:`, that will no longer be possible and we will have to update #3255 to handle this new use case.